diff --git a/drivers/led/led_gpio.c b/drivers/led/led_gpio_cpy.c
index ddfaa3456b..aa3ce16c4a 100644
--- a/drivers/led/led_gpio.c
+++ b/drivers/led/led_gpio_cpy.c
@@ -11,6 +11,7 @@
 #include <led.h>
 #include <asm/gpio.h>
 #include <dm/lists.h>
+#include <dm/uclass-internal.h>
 
 DECLARE_GLOBAL_DATA_PTR;
 
@@ -126,6 +127,14 @@ static int led_gpio_bind(struct udevice *parent)
 			return ret;
 		uc_plat = dev_get_uclass_platdata(dev);
 		uc_plat->label = label;
+
+		if (ofnode_read_bool(node, "default-state")) {
+			struct udevice *devp;
+
+			ret = uclass_get_device_tail(dev, 0, &devp);
+			if (ret)
+				return ret;
+
 	}
 
 	return 0;
diff --git a/drivers/led/led_gpio.c b/drivers/led/led_gpio_cpy.c
index aa3ce16c4a..7f7e967166 100644
--- a/drivers/led/led_gpio.c
+++ b/drivers/led/led_gpio_cpy.c
@@ -134,6 +134,7 @@ static int led_gpio_bind(struct udevice *parent)
 			ret = uclass_get_device_tail(dev, 0, &devp);
 			if (ret)
 				return ret;
+		}
 
 	}
 
